com.unity.formats.fbx/Editor/ConvertToNestedPrefab.cs

```diff
@@ -526,6 +535,56 @@ internal static List<GameObject> GetSceneReferencesToObject(Object obj)
             return prefab;
         }
 
+        internal struct SourceObjectInfo
+        {
+            public GameObject destGO;
+#if UNITY_2021_2_OR_NEWER
+            public List<Object> sceneObjectsWithReference;
+#endif // UNITY_2021_2_OR_NEWER
+
+            public SourceObjectInfo(GameObject dest)
+            {
+                destGO = dest;
+#if UNITY_2021_2_OR_NEWER
+               sceneObjectsWithReference = new List<Object>();
+#endif // UNITY_2021_2_OR_NEWER
+        }
+    }
+
+        internal static Dictionary<string, SourceObjectInfo> s_nameToInfo = new Dictionary<string, SourceObjectInfo>();
+
+        internal static void GatherSceneHierarchy(GameObject hierarchyRoot)
+        {
+            s_nameToInfo.Clear();
+
+#if UNITY_2021_2_OR_NEWER
+            var isPreviewScene = SceneManagement.EditorSceneManager.IsPreviewSceneObject(hierarchyRoot);
+#endif // UNITY_2021_2_OR_NEWER
+
+            var q = new Queue<Transform>();
+            q.Enqueue(hierarchyRoot.transform);
+            while (q.Count > 0)
+            {
+                var t = q.Dequeue();
+
+                var info = new SourceObjectInfo();
+#if UNITY_2021_2_OR_NEWER
+                if (!isPreviewScene)
+                {
+                    var instanceID = t.gameObject.GetInstanceID();
+                    info.sceneObjectsWithReference = GetSceneReferences(instanceID);
```

Comment:
This means: for all objects t that are being converted, loop over the entire scene to get the references to t.

Better than before when it was for all *components* loop over the entire scene but still quadratic.

If this is fast enough then we can leave it there but we could theoretically do better.

Reply:
I think I could make it so that it does one huge query but then it will be more difficult to figure out which component reference needs to be fixed in which result.